x-pack/plugins/ml/public/application/components/delete_job_check_modal/delete_job_check_modal.tsx

```diff
@@ -0,0 +1,234 @@
+      <FormattedMessage
+        id="xpack.ml.deleteJobCheckModal.modalTextClose"
+        defaultMessage="{ids} cannot be deleted and cannot be removed from current space."
+        values={{ ids: jobIds.join(', ') }}
+      />
+    ),
+  };
+
+  if (canDelete) {
+    return {
+      buttonText: (
+        <FormattedMessage
+          id="xpack.ml.deleteJobCheckModal.buttonTextCanDelete"
+          defaultMessage={`Continue to delete {length, plural, one {# job} other {# jobs}}`}
+          values={{ length: jobIds.length }}
+        />
+      ),
+      modalText: (
+        <FormattedMessage
+          id="xpack.ml.deleteJobCheckModal.modalTextCanDelete"
+          defaultMessage="{ids} can be deleted."
+          values={{ ids: jobIds.join(', ') }}
+        />
+      ),
+    };
+  } else if (!canDelete && canUnTag) {
+    return {
+      buttonText: (
+        <FormattedMessage
+          id="xpack.ml.deleteJobCheckModal.buttonTextCanUnTagConfirm"
+          defaultMessage={`Remove {length, plural, one {# job} other {# jobs}} from space`}
+          values={{ length: jobIds.length }}
+        />
+      ),
+      modalText: (
+        <FormattedMessage
+          id="xpack.ml.deleteJobCheckModal.modalTextCanUnTag"
+          defaultMessage="{ids} cannot be deleted but can be removed from the current space."
+          values={{ ids: jobIds.join(', ') }}
+        />
+      ),
+    };
+  } else if (!canDelete && !canUnTag) {
+    return noActionContent;
+  }
+
+  return noActionContent;
+}
+
+interface Props {
+  canDeleteCallback: () => void;
+  onCloseCallback: () => void;
+  jobType: JobType;
+  jobIds: string[];
+}
+
+export const DeleteJobCheckModal: FC<Props> = ({
+  canDeleteCallback,
+  onCloseCallback,
+  jobType,
+  jobIds,
+}) => {
+  const [buttonContent, setButtonContent] = useState<JSX.Element | undefined>();
+  const [modalContent, setModalContent] = useState<JSX.Element | undefined>();
+  const [shouldUnTag, setShouldUnTag] = useState<boolean>(false);
+  const [jobCheckRespSummary, setJobCheckRespSummary] = useState<JobCheckRespSummary | undefined>();
+
+  const {
+    savedObjects: { canDeleteJob },
+  } = useMlApiContext();
+  const { displayErrorToast } = useToastNotificationService();
+
+  useEffect(() => {
+    // Do the spaces check and set the content for the modal and buttons depending on results
+    canDeleteJob(jobType, jobIds).then((resp) => {
+      const respSummary = getRespSummary(resp);
+      setJobCheckRespSummary(respSummary);
+      const { buttonText, modalText } = getModalContent(jobIds, respSummary);
+      setButtonContent(buttonText);
+      setModalContent(modalText);
+    });
+  }, []);
+
+  const onClick = async () => {
+    // Remove job(s) from current space then execute relevant callback
+    // TODO: get current space for job
+    const currentSpace = 'default';
+    if (shouldUnTag === true) {
+      const resp = await ml.savedObjects.removeJobFromSpace(jobType, jobIds, [currentSpace]);
```

Comment:
`removeJobFromSpace` can be retrieved from `useMlApiContext` (line 159 ^^)
`removeJobFromCurrentSpace` is now available and should be used instead of `removeJobFromSpace`